mpitests/test_adjoint_jacobian.py

```diff
@@ -1381,3 +1381,104 @@ def circuit(params):
     comm.Barrier()
 
     assert np.allclose(j_cpu, j_gpu)
+
+
+@pytest.mark.parametrize("n_targets", [1, 2])
+def test_qubit_unitary(n_targets):
+    """Tests that ``qml.QubitUnitary`` can be included in circuits differentiated with the adjoint method."""
+    np.random.seed(1337)
+    n_wires = 6
+    dev = qml.device("lightning.gpu", wires=n_wires, mpi=True)
```

Comment:
How about adding tests for FP32?

Reply:
Nice catch. It wasn't trivial to make float32 work with my MPI tests. All good now.